go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -175,25 +178,43 @@ func (oc *DefaultNetworkController) syncEgressFirewall(egressFirewalls []interfa
 		}
 		existingEFNamespaces[ef.Namespace] = true
 	}
+	// move ACLs from cluster port group to per-namespace port group
+	// old acl matches on the address set, which is still valid. match change form address set to port group will be
+	// performed by the egress firewall handler.
+	// sync the ovn and k8s egressFirewall states.
 	predicateIDs := libovsdbops.NewDbObjectIDs(libovsdbops.ACLEgressFirewall, oc.controllerName, nil)
 	aclP := libovsdbops.GetPredicate[*nbdb.ACL](predicateIDs, nil)
 	efACLs, err := libovsdbops.FindACLsWithPredicate(oc.nbClient, aclP)
 	if err != nil {
 		return fmt.Errorf("cannot find Egress Firewall ACLs: %v", err)
 	}
-	deleteACLs := []*nbdb.ACL{}
-	for _, efACL := range efACLs {
-		if !existingEFNamespaces[efACL.ExternalIDs[libovsdbops.ObjectNameKey.String()]] {
-			// there is no egress firewall in that namespace, delete
-			deleteACLs = append(deleteACLs, efACL)
+	var ops []libovsdb.Operation
+	err = batching.Batch[*nbdb.ACL](aclUpdateBatchSize, efACLs, func(batchACLs []*nbdb.ACL) error {
+		for _, batchACL := range batchACLs {
+			namespace := batchACL.ExternalIDs[libovsdbops.ObjectNameKey.String()]
+			if namespace != "" {
+				pgName := oc.getNamespacePortGroupName(namespace)
+
+				// delete stale ACLs from namespaced port group
+				if !existingEFNamespaces[namespace] {
+					ops, err = libovsdbops.DeleteACLsFromPortGroupOps(oc.nbClient, ops, pgName, batchACL)
+				} else {
+					// re-attach from ClusterPortGroupNameBase to namespaced port group.
+					// port group should exist, because namespace handler will create it.
+					ops, err = libovsdbops.AddACLsToPortGroupOps(oc.nbClient, ops, pgName, batchACL)
+				}
+				// delete all EF ACLs from ClusterPortGroupNameBase
+				ops, err = libovsdbops.DeleteACLsFromPortGroupOps(oc.nbClient, ops,
+					oc.getClusterPortGroupName(types.ClusterPortGroupNameBase), batchACL)
```

Comment:
I would accumulate in slices as before and use less ops

Reply:
maybe not that straight forward if it needs to be accumulated by namespace, but probably still doable and worth it?

Reply:
yep I think its not that complicated